### PR TITLE
Shadowling Balances #3

### PR DESCRIPTION
**Pull request by ghost**
:cl: Thefastfoodguy
refactor: Refactored shadowlings and thralls to use datum antagonists
balance: Shadowlings now will, while in the darkness, generate shield charges, up to a maximum of 3, which can absorb projectiles!
balance: Shadowling Glare now deals more stamina damage
tweak: Shadowlings using commune now have slightly bigger text than thralls
balance: Silicons who have their lights veiled by a shadowling now need to have them fixed by a Roboticist (or really, any guy with tools)
fix: Shadow walk works again
/:cl:

### CODE REVIEW

**Comment by Manly-Dwarf** [comment]
so it's shadowling commune but dumber

eh, whatever

File: hippiestation/code/datums/antagonists/shadowling.dm [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+			M.gib()
+
+/datum/antagonist/shadowling/greet()
+	to_chat(owner, "<br> <span class='shadowling'><b><font size=3>You are a shadowling!</font></b></span>")
+	to_chat(owner, "<b>Currently, you are disguised as an employee aboard [station_name()]].</b>")
+	to_chat(owner, "<b>In your limited state, you have three abilities: Enthrall, Hatch, and Hivemind Commune.</b>")
+	to_chat(owner, "<b>Any other shadowlings are your allies. You must assist them as they shall assist you.</b>")
+	to_chat(owner, "<b>If you are new to shadowling, or want to read about abilities, check the wiki page at https://wiki.hippiestation.com/index.php?title=Shadowling</b><br>")
+	to_chat(owner, "<b>You require [SSticker.mode.required_thralls || 15] thralls to ascend.</b><br>")
+	to_chat(owner, "<b>As a shadowling, you can naturally speak Shadowtounge (,8). This is a useful <i>private</i> way to communicate with allies.</b>")
```
Comment (Manly-Dwarf) [inline hippiestation/code/datums/antagonists/shadowling.dm]: "shadowtounge"?
tongue

File: hippiestation/code/datums/antagonists/thrall.dm [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+						  under their command...</span>")
+	return TRUE
+
+/datum/antagonist/thrall/greet()
+	to_chat(owner, "<span class='shadowling'><b>You see the truth. Reality has been torn away and you realize what a fool you've been.</b></span>")
+	to_chat(owner, "<span class='shadowling'><b>The shadowlings are your masters.</b> Serve them above all else and ensure they complete their goals.</b></span>")
+	to_chat(owner, "<span class='shadowling'>You may not harm other thralls or the shadowlings. However, you do not need to obey other thralls.</span>")
+	to_chat(owner, "<span class='shadowling'>Your body has been irreversibly altered. The attentive can see this - you may conceal it by wearing a mask.</span>")
+	to_chat(owner, "<span class='shadowling'>Though not nearly as powerful as your masters, you possess some weak powers. These can be found in the Thrall Abilities tab.</span>")
+	to_chat(owner, "<span class='shadowling'>You may communicate with your allies by using the Lesser Commune ability, or by speaking to them in the Shadowtounge (,8).</span>")
```
Comment (Manly-Dwarf) [inline hippiestation/code/datums/antagonists/thrall.dm]: v

**Comment by ghost** [comment]
Either make it so shadowlings talk in their own language, or use the commune powers.
Doing both is silly.

**Comment by ghost** [comment]
@McBawbaggings Truth be told, the language is just for yelling at thralls/other slings in person

**Comment by ghost** [comment]
Choose either the language or commune.
Not both.
I'd prefer commune.
Remove the bullet dodging. That's just excessive.
Everything else seems ok.

**Comment by ghost** [comment]
@Dudeinyourbackyard Why remove the bullet dodging? It basically forces you to actually bring the light to the slings to kill them with bullets

**Comment by ghost** [comment]
Anyways I removed shadowtounge

**Comment by ghost** [comment]
ok
+1

File: hippiestation/code/datums/antagonists/shadowling.dm [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+			count++
+
+	if(LAZYLEN(SSticker.mode.shadows))
+		parts += printplayerlist(SSticker.mode.shadows)
+
+	return "<div class='panel redborder'>[parts.Join("<br>")]</div>"
+
+
+/datum/objective/ascend
+	explanation_text = "Ascend to your true form by use of the Ascendance ability. This may only be used with 15 or more collective thralls, while hatched, and is unlocked with the Collective Mind ability."
```
Comment (GuyonBroadway) [inline hippiestation/code/datums/antagonists/shadowling.dm:99]: You say however many thralls necessary scales, might wanna update this line.

Comment (ghost) [reply hippiestation/code/datums/antagonists/shadowling.dm:99]: It IS updated, see update_explanation_text()

**Comment by Manly-Dwarf** [comment]
Why do shadowlings need to be immune to taser/bullet shots?
Are you absolutely insane?

**Comment by ghost** [comment]
@Kawaii-Big-Boss I changed that, basically they now have a "shadow shield", which has a max of 3 charges, which recharges in ```40 - amount of thralls``` seconds. If the shield has charges, it dodges.

**Comment by Manly-Dwarf** [comment]
But why? Shadowlings are boring and annoying enough to fight against.

**Comment by ghost** [comment]
@Kawaii-Big-Boss the point of shadowlings is that you have to force them into some light to fight them. They don't regenerate the shielding in light. Also the shielding doesn't affect you toolboxing / harmbatoning / crowbaring them to death, only projectiles.

**Comment by Manly-Dwarf** [comment]
But normally when a Shadowling encounters a guy who is going to shoot at/attack them they just run away anyhow?
Why do they need a shield?

**Comment by ghost** [comment]
@Kawaii-Big-Boss As long as you shoot it 3 times, it'll be super vulnerable. Maximum of 3 shield charges.

Also because shadowlings have been super weak since I nerfed glare. I haven't seen one win for em.

**Comment by ghost** [comment]
@HippieStation/coders @HippieStation/maintainers this needs review

@HippieStation/head this also needs one more +1

**Review by ghost — DISMISSED** [review]
Code seems OK

**Comment by JohnGinnane** [comment]
Need more votes @HippieStation/head  also @MoreRobustThanYou you've got a conflict((

**Comment by GuyonBroadway** [comment]
MRTY isn't gonna fix this cus erik banned them under rule 0 due to their bad in-game behaviour.

**Comment by MoreRobustThanYou** [comment]
Bah; humbug. I'll fix the conflict, but Erik can go fuck himself sideways with a hooker for all I care.

**Comment by GuyonBroadway** [comment]
Now travis hates you.

**Comment by JohnGinnane** [comment]
@GuyonBroadway slings are b&?

**Comment by GuyonBroadway** [comment]
Not to my knowledge

**Comment by JohnGinnane** [comment]
Oh sorry you mean MRTY is banned, my bad

**Comment by ghost** [comment]
should be ready now

**Comment by ghost** [comment]
Should be fine to merge if someone tests it

**Comment by ghost** [comment]
yes i tested this REEE

**Comment by ghost** [comment]
Not you, a maintainer

**Comment by praisenarsie** [comment]
sorry who

also we have testmerges for a reason erik

**Comment by ghost** [comment]
@HippieStation/maintainers test/merge this already

**Comment by praisenarsie** [comment]
we don't have remote. only headmins and host (erik) do.

also nice job trying to tag people you can't tag anymore 😉

**Comment by ghost** [comment]
i didn't mean testmerge, i meant test and then merge

**Comment by praisenarsie** [comment]
😠

**Comment by Pyko1** [comment]
would prolly be easist to testmerge tho?

**Comment by Carbonhell** [comment]
Please do not mix refactors/bugfixes with balance/features ree
I dont think testmerge is viable so someome has to pull the code in local and test manually,or this will be denied in 1 week from now

**Comment by MoreRobustThanYou** [comment]
So wait: this might be denied if the maintainers are too lazy to test it? HAHAHAHAHAHAHAHA

**Comment by praisenarsie** [comment]
@MoreRobustThanYou stop acting like a fucking tool, if you knew how to fucking test stuff properly we wouldn't be here.

@Carbonhell i very much doubt any maintainer here wants to test this, we barely use testmerges anyway, it's like three fucking clicks to testmerge it either way.
also who cares if balances are mixed in with this refactor, slings suck weewee anyway.

**Comment by Carbonhell** [comment]
If balances werent mixed,if they are shit they can be reverted easily

**Comment by praisenarsie** [comment]
also note, when i mean they suck weewee i mean they suck in the creative aspect.

**Comment by ghost** [comment]
@Chnkr testmerge this

**Comment by ghost** [comment]
stale

**Comment by ghost** [comment]
@Chnkr you mean you have a grudge against me and you are too lazy to get off your ass and do 10 minutes of testing? go fuck yourself with a cactus, you prick

**Comment by Manly-Dwarf** [comment]
We also don't have Shadowlings in rotation and nobody really wants it. Sorry, man - but you devoted way too much time and passion into a project that nobody liked.

**Comment by ghost** [comment]
Oh god, you idiots removed it?

**Comment by KayeArray** [comment]
unknown-25

**Comment by Manly-Dwarf** [comment]
We gained control over our config again, four gamemodes were killed because nobody liked them - except one of them, which we're discussing re-adding with a couple of changes.

Besides, I think it's a bit late to complain about removing gamemodes, Amari "remove gangs from upstream" Ray.

**Comment by ghost** [comment]
gangs was the exception -- it was cancer

**Comment by Manly-Dwarf** [comment]
Yeah dude, it was so cancer that everyone on the server complains about it being gone.

Oh wait, maybe you just didn't like it and threw a hissy over the fact that everyone called you retarded. Yeah, that sounds more likely.

**Comment by Pyko1** [comment]
@MoreRobustThanYou it was disabled from config due to https://github.com/HippieStation/HippieStation/pull/6031until someone makes shadowling thinggys for new antagpanel.

**Comment by ghost** [comment]
@Pyko1 you dipshit, this adds antag datums for slings, which would add it to the new panel

**Comment by ghost** [comment]
@Kawaii-Big-Boss hippie players are usually retarded anyways, no surprise there.

**Comment by Manly-Dwarf** [comment]
And yet you question why we banned you for being a malicious coder who cared only for themselves?

Sure hope you enjoy hanging around catgirl faggots from now and on.

**Comment by ghost** [comment]
I was banned for posting a stupid exploit in OOC, and also because 90% of the admins were corrupt, and retarded enough to think that i purposely fucked up hotkeys

**Comment by Carbonhell** [comment]
if you can't be assed to even do the basic test of turning a server on and see if atleast normal movement works it's not our fault nigga

**Comment by ghost** [comment]
@Carbonhell Actually, a git merge failed somewhere, which I think caused MANY files to be left out, fucking things up

**Comment by ghost** [comment]
Incase you nitwits were too dumb to notice, I tried to fix this

**Comment by Manly-Dwarf** [comment]
Amari goes nuclear

**Comment by ghost** [comment]
and honestly, I have no plans to try to ban appeal again on here, until there is a better host and a a box that isn't a potato

**Comment by Manly-Dwarf** [comment]
Good news! We have a new host coming up with a brand new box that isn't a potato. His name is Erik, I'm sure you'll come to like him. He codes, just like you!

**Comment by ghost** [comment]
Is erik a bit less of a prick now?

**Comment by Carbonhell** [comment]
no

**Comment by Manly-Dwarf** [comment]
Don't listen to Carbonhell, she's just Erik's bitch.

**Comment by ghost** [comment]
I think we are done here, feel free to open a new PR if you want to continue on this

**Pull request closed without merging**